pyomo/repn/tests/ampl/test_ampl_comparison.py

```diff
@@ -11,127 +11,143 @@
 # Test the Pyomo NL writer against the AMPL NL writer
 #
 
+import json
 import re
 import glob
 import subprocess
 import os
 from os.path import abspath, dirname
 currdir = dirname(abspath(__file__))+os.sep
 
-import pyutilib.th as unittest
+import pyomo.common.unittest as unittest
+import pyomo.common
 
 import pyomo.scripting.pyomo_main as main
 
+parameterized, param_available = pyomo.common.dependencies.attempt_import('parameterized')
+if not param_available:
+    raise unittest.SkipTest('Parameterized is not available.')
+
 # https://github.com/ghackebeil/gjh_asl_json
 has_gjh_asl_json = False
 if os.system('gjh_asl_json -v') == 0:
     has_gjh_asl_json = True
 
+names = []
+# add test methods to classes
+for f in glob.glob(currdir+'*_testCase.py'):
+    names.append(re.split('[._]',os.path.basename(f))[0])
+
+
 class Tests(unittest.TestCase):
 
     def pyomo(self, cmd):
         os.chdir(currdir)
         output = main.main(['convert', '--logging=quiet', '-c']+cmd)
         return output
 
+
 class BaselineTests(Tests):
     def __init__(self, *args, **kwds):
         Tests.__init__(self, *args, **kwds)
 
-#
-#The following test generates an nl file for the test case
-#and checks that it matches the current pyomo baseline nl file
-#
-@unittest.nottest
-def nlwriter_baseline_test(self, name):
-    if os.path.exists(currdir+name+'.dat'):
-        self.pyomo(['--output='+currdir+name+'.test.nl',
-                    currdir+name+'_testCase.py',
-                    currdir+name+'.dat'])
-    else:
-        self.pyomo(['--output='+currdir+name+'.test.nl',
-                    currdir+name+'_testCase.py'])
-
-    # Check that the pyomo nl file matches its own baseline
-    self.assertFileEqualsBaseline(
-        currdir+name+'.test.nl', currdir+name+'.pyomo.nl',
-        tolerance=(1e-7, False))
+    #
+    #The following test generates an nl file for the test case
+    #and checks that it matches the current pyomo baseline nl file
+    #
+    @parameterized.parameterized.expand(input=names)
+    def nlwriter_baseline_test(self, name):
+        if os.path.exists(currdir+name+'.dat'):
+            self.pyomo(['--output='+currdir+name+'.test.nl',
+                        currdir+name+'_testCase.py',
+                        currdir+name+'.dat'])
+        else:
+            self.pyomo(['--output='+currdir+name+'.test.nl',
+                        currdir+name+'_testCase.py'])
+
+        # Check that the pyomo nl file matches its own baseline
+        with open(currdir+name+'.test.nl', 'r') as f1, \
+                open(currdir+name+'.pyomo.nl', 'r') as f2:
+                    f1_contents = list(filter(None, f1.read().replace('n', 'n ').split()))
+                    f2_contents = list(filter(None, f2.read().replace('n', 'n ').split()))
+                    for item1, item2 in zip(f1_contents, f2_contents):
```

Comment:
This should probably use `itertools.zip_longest` so that differing length files cause errors.